src/core/pointcloud/qgspointcloudlayer.cpp

```diff
@@ -67,8 +67,10 @@ QgsPointCloudLayer::QgsPointCloudLayer( const QString &uri,
 
 QgsPointCloudLayer::~QgsPointCloudLayer()
 {
-  // TODO: switch to using private resources for the inner workings of the stats calculation task
-  waitForStatisticsCalculationToFinish( true );
+  if ( QgsTask *task = QgsApplication::taskManager()->task( mStatsCalculationTask ) )
+  {
+    task->cancel();
+  }
```

Comment:
Is QgsTask::cancel() really *waiting* for the thread to complete/terminate ? I'm asking because it looks like the race condition in eptprovider is back and it could be due to this change

Reply:
task->cancel() should not be waiting, and that's okay, because the stats calculation task should not need anything from QgsPointCloudLayer (so the layer can be deleted safely while the task is still getting canceled) - or at least that's how it is meant to work...